### DISCUSSION TITLE
How to use the debugger in VSCode?

### DISCUSSION THREAD

**Discussion by clizbe** [TulipaEnergy/TulipaEnergyModel.jl#84] [Questions]
I tried running the debugger (to see if I could peak inside variables while the code is running) and it crashed and gave me this error:

![image](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/assets/11889283/851d584d-4343-468a-846c-f5d1a607c99b)

I had set a breakpoint on one line (a red circle), but I don't think that's the problem.

**Comment by abelsiqueira** [comment]
I can reproduce.
I managed to make it work by running it from the runtests.jl file. I.e., you can set a breakpoint in a different file, but then open runtests.jl and run the debugger there.
I think the solution is running a script (an entry point running the code, not just defining it) instead of the code.
It should probably be possible to define a debugger configuration that states what file should be run when the debugger is activated. I haven't done it in a while, so I'm not sure for now.

**Comment by clizbe** [comment]
Good thing we had a course on this!
